### DISCUSSION TITLE
how to make filter that hidden form user but is running by default????

### DISCUSSION THREAD

**Discussion by Rizkommam111** [filamentphp/filament#9094] [Help]
so i'am building cms for my e-commerce project but i want to make it so that only certain user can see certain data, so i'am trying using filter that i set it default, but also i want to set it hidden so that the user cant change it but everytime i set it hidden the default function also stopped. any solution is also accepted

**Comment by VincvanAs** [comment]
Could you share some code?

**Reply by Rizkommam111** [reply]
`<?php

class UserResource extends Resource
{
    protected static ?string $model = User::class;

    protected static ?string $navigationIcon = 'heroicon-o-rectangle-stack';

    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\TextInput::make('name')
                    ->required()
                    ->default(auth()->user()->id)
                    ->maxLength(255),
                Forms\Components\TextInput::make('email')
                    ->email()
                    ->required()
                    ->maxLength(255),
                Forms\Components\DateTimePicker::make('email_verified_at'),
                Forms\Components\TextInput::make('password')
                    ->password()
                    ->required()
                    ->maxLength(255),
            ]);
    }

    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('name')
                    ->searchable(),
                Tables\Columns\TextColumn::make('email')
                    ->searchable(),
                Tables\Columns\TextColumn::make('email_verified_at')
                    ->dateTime()
                    ->sortable(),
                Tables\Columns\TextColumn::make('created_at')
                    ->dateTime()
                    ->sortable()
                    ->toggleable(isToggledHiddenByDefault: true),
                Tables\Columns\TextColumn::make('updated_at')
                    ->dateTime()
                    ->sortable()
                    ->toggleable(isToggledHiddenByDefault: true),
            ])
            ->filters([
                Tables\Filters\Filter::make('id')
                    ->query(fn (Builder $query): Builder => $query->where('id', auth()->user()->id))
                    ->default()
                    ->hidden()
            ])
            ->actions([
                Tables\Actions\EditAction::make(),
            ])
            ->bulkActions([
                Tables\Actions\BulkActionGroup::make([
                    Tables\Actions\DeleteBulkAction::make(),
                ]),
            ]);
    }
    
    public static function getRelations(): array
    {
        return [
            //
        ];
    }
    
    public static function getPages(): array
    {
        return [
            'index' => Pages\ListUsers::route('/'),
            'create' => Pages\CreateUser::route('/create'),
            'edit' => Pages\EditUser::route('/{record}/edit'),
        ];
    }    
}
`

so im using this code from my User resource as testing so that my main code dont get messed up

**Comment by Rizkommam111** [comment]
i have found the solution apparently i have to use "modofyQueryusing" function lol

**Reply by mahmudz** [reply]
Been searching for an hour. It's a life saver.

But it's weird that other example the doc suggesting(->default()) is not working.
